### PR TITLE
Refactor signing provider get key from destination

### PR DESCRIPTION
**Pull request by sieniven**
## Summary

- Rename GetKeyForDestination to GetKeyOrDefaultFromDestination
- Refactor GetKeyOrDefaultFromDestination to use FromOrDefaultDestionation with new signing provider key type

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [x] None

### CODE REVIEW

**Comment by prasannavl** [comment]
Looks good. Further, could also abstract the flag checks and remove all the & with a combined singular mask.

**Review by prasannavl — DISMISSED** [review]

File: src/script/signingprovider.cpp [hunk 1]
```diff
@@ -186,28 +186,29 @@ bool FillableSigningProvider::GetCScript(const CScriptID &hash, CScript& redeemS
...
+            break;
+        case WitV16KeyEthHashType:
+            id.type = KeyAddressType::UNCOMPRESSED;
+            break;
+        case ScriptHashType: {
+            CScript script;
+            CScriptID script_id(id);
+            CTxDestination inner_dest;
+            if (store.GetCScript(script_id, script) && ExtractDestination(script, inner_dest)) {
+                id = CKeyID::FromOrDefaultDestination(inner_dest,KeyType::EthHashKeyType);
```
Comment (prasannavl) [inline src/script/signingprovider.cpp:206]: `            if (auto inner_witness_id = std::get_if<WitnessV0KeyHash>(&inner_dest)) {`

It's checking for V0KeyHash?

File: src/script/standard.h [hunk 2]
```diff
@@ -163,10 +163,11 @@ enum KeyType {
     PKHashKeyType = 1 << 0,
     ScriptHashKeyType = 1 << 1,
     WPKHashKeyType = 1 << 2,
-    EthHashKey = 1 << 3,
+    EthHashKeyType = 1 << 3,
```
Comment (prasannavl) [inline src/script/standard.h:166]: We're using ERC55 terminology now.

File: src/script/signingprovider.cpp (hunk 1 above)
Comment (sieniven) [reply src/script/signingprovider.cpp:206]: Gosh sorry for this and thanks for the spot.

File: src/script/standard.h (hunk 2 above)
Comment (prasannavl) [reply src/script/standard.h:166]: We can rename the others as well to use the same terminology (in a separate PR), but let's move that direction, than this direction :)

Comment (sieniven) [reply src/script/standard.h:166]: Got it. We can do the fix for this in a separate PR.

File: src/script/standard.h [hunk 3]
```diff
@@ -163,10 +163,11 @@ enum KeyType {
...
     MNOperatorKeyType = (1 << 4) | PKHashKeyType | WPKHashKeyType,
     MNOwnerKeyType = (1 << 5) | PKHashKeyType | WPKHashKeyType,
     MNRewardKeyType = (1 << 6) | PKHashKeyType | ScriptHashKeyType | WPKHashKeyType,
+    SigningProviderType = (1 << 7) | PKHashKeyType | ScriptHashKeyType | WPKHashKeyType | EthHashKeyType,
```
Comment (prasannavl) [inline src/script/standard.h:170]: Note: These should just be combinations, Providing it's own values will break many bit flag assumputions

File: src/pubkey.h [hunk 4]
```diff
@@ -36,31 +36,22 @@ class CKeyID : public uint160
...
-                break;
-            case WitV16KeyEthHashType:
-                if ((filter & KeyType::EthHashKeyType) == KeyType::EthHashKeyType) {
-                    return CKeyID(std::get<WitnessV16EthHash>(dest));
-                }
-                break;
-            default: 
-                return {};
+        KeyType destType = FromOrDefaultDestinationTypeToKeyType(dest.index());
+        if ((destType & filter) == KeyType::PKHashKeyType) {
```
Comment (prasannavl) [inline src/pubkey.h:40]: Let's hoist the filtering up and why not just use a switch?

Comment (sieniven) [reply src/pubkey.h]: Resolved in [f7e7db6](https://github.com/DeFiCh/ain/pull/2259/commits/f7e7db6411a7a5175c58cad5f7ce731e5c0ce4cd)

**Review by prasannavl — DISMISSED** [review]

File: src/pubkey.h [hunk 5]
```diff
@@ -36,31 +36,19 @@ class CKeyID : public uint160
     static std::optional<CKeyID> TryFromDestination(const CTxDestination &dest, KeyType filter=KeyType::UnknownKeyType) {
         // Explore switching TxDestType to a flag type. Then, we can easily take an allowed
```
Comment (prasannavl) [inline src/pubkey.h:37]: Can remove these comments now

**Review by Bushstar — DISMISSED** [review]

**Pull request merged**